packages/server/src/sync/service.ts

```diff
@@ -198,9 +199,16 @@ export class SyncService extends Service {
       const oldWebhookIndex = oldWebhooks.findIndex((x) => x.url === webhook.url)
 
       if (oldWebhookIndex < 0) {
-        const token = await this.webhooks.generateToken()
-        const newToken = await this.webhooks.create(clientId, token, webhook.url)
-        webhooksWithTokens.push({ url: newToken.url, token: newToken.token })
+        try {
+          await axios.options(webhook.url)
+
+          const token = await this.webhooks.generateToken()
+          const newToken = await this.webhooks.create(clientId, token, webhook.url)
+          webhooksWithTokens.push({ url: newToken.url, token: newToken.token })
+        } catch {
+          // TODO: Send this info back to the user
+          this.logger.warn(`Webhook with url "${webhook.url}" cannot be reached. Skipping...`)
+        }
```

Comment:
Yup we need to return a 400 error with that messages. The sync shouldn't happen if the webhook is unreachable otherwise it kind of confusing to have the webhooks be scretly not set.

Reply:
So to avoid other changes being done, we'd need to have a verify webhooks function before everything else.

Reply:
Since we seem to sync bots before the /receive API is up, I had to remove this validation! We will simply make sure that the URL sent is of valid format using Joi.